go/tosca/interpreter_registry.go

```diff
@@ -48,22 +81,43 @@ func GetAllRegisteredInterpreters() map[string]Interpreter {
 // and a panic is triggered if an implementation was bound to the same name
 // before, or the implementation is nil. This function is mainly intended to be
 // used by package initialization code.
+//
+// Deprecated: Use RegisterInterpreterFactory instead.
 func RegisterInterpreter(name string, impl Interpreter) {
+	err := RegisterInterpreterFactory(name, func(any) (Interpreter, error) {
+		return impl, nil
+	})
+	if err != nil {
+		panic(err)
+	}
+}
+
+// RegisterInterpreterFactory registers a new Interpreter implementation
+// to be exported for general use in the binary. The name is not case-sensitive,
+// and a panic is triggered if a factory was bound to the same name before, or
+// the factory is nil. This function is mainly intended to be used by package
```

Comment:
if it is meant as package initialization, should it be public ? should it not be in `func init()` ?

Reply:
This is meant to be a function to be **used** by package initialization code. It is not initializing this package.